### DISCUSSION TITLE
Multiple Lockable Node / Import / Inspector Tabs

### DISCUSSION THREAD

**Discussion by mymstake** [godotengine/godot-proposals#5242] [Editor]
Godot 4.0's Inspector, Node, Scene, Import, and File System tabs have received some wonderful and welcome improvements.  Among them: we can now un-dock these tabs!  I find this especially wonderful 👍🏻

Now feels like the proper time to suggest locked tabs -more specifically, all except the "Scene" tab)

One can look to Unity for a prime example of exactly how I imagine this feature implementation:

- Up to 4 [inspectors] can be open at once.
- [Inspectors] can be locked to prevent [node] selections from changing their focus.
- Switching between open [scenes] returns each [inspector] to its last status in that [scene]
- Changing a [node]'s properties in a locked [inspector] does not focus that [node]

I haven't used Unity in a while, but I can imagine that with the ability to select multiple nodes and edit any properties they share simultaneously, these batch selections should also be lockable to an inspector.  This would be extremely useful saving time selecting multiple specific nodes of very different types repeatedly.  Grouping them in another node is not always an option)

Related Thoughts:

- The output/debugging/graphing/tileset/animation drawer.  I forgot to check if it still (as of 4.0 alpha 14) automatically jumps to certain tabs when specific nodes -mostly animation based nodes- are focused, but this behavior would need handling as there's no outright answer to whether a user would want the drawer to auto-focus the most recent node with a relevant tab or stay locked to the last one selected in a locked inspector.
- The **inspector selection history** (not undo/redo history) is great and I use it a lot and it's really small and could use some love and attention.  Perhaps integrating the locking into the history feature or visa-versa would help highlight both of them to users.
- The Node (Signals/Groups) and Import tabs could also benefit from some kind of multiple instances / locking, but I personally place that feature at such lower relative priority I nearly left it out.

**Comment by arkology** [comment]
Related to https://github.com/godotengine/godot-proposals/discussions/12709
